pulp_python/app/tasks/repair.py

```diff
@@ -0,0 +1,85 @@
+import os
+from gettext import gettext as _
+import logging
+
+import uuid
+from django.conf import settings
+
+from pulp_python.app.models import PythonPackageContent, PythonRepository
+from pulp_python.app.utils import artifact_to_python_content_data
+
+log = logging.getLogger(__name__)
+
+
+def repair(repository_pk: uuid.UUID) -> None:
+    """
+    Repairs metadata of all packages for the specified repository.
+
+    Args:
+        repository_pk (uuid.UUID): The primary key of the repository to repair.
+
+    Returns:
+        None
+    """
+    repository = PythonRepository.objects.get(pk=repository_pk)
+
+    log.info(
+        _(
+            "Repairing packages' metadata for the latest version of repository {}."
+        ).format(repository.name)
+    )
+    content_set = repository.latest_version().content.values_list("pk", flat=True)
+    content = PythonPackageContent.objects.filter(pk__in=content_set)
+    num_repaired = repair_metadata(content)
+    log.info(
+        _("{} packages processed, {} package metadata repaired.").format(
+            len(content_set), num_repaired
+        )
+    )
```

Comment:
Let's use ProgressReports (https://github.com/pulp/pulpcore/blob/main/pulpcore/app/models/progress.py#L19) to report on the number of packages repaired.